ci/scripts/java_jni_macos_build.sh

```diff
@@ -41,9 +41,15 @@ install_dir=${build_dir}/cpp-install
 : ${ARROW_PLASMA:=ON}
 : ${ARROW_PYTHON:=OFF}
 : ${ARROW_S3:=ON}
+: ${ARROW_USE_CCACHE:=OFF}
```

Comment:
Why not, but you should perhaps also pass it to CMake below?

Reply:
Ah, I changed wrong file... I should have changed `java_jni_manylinux_build.sh`...

I've passed it to CMake arguments. I've updated `java_jni_manylinux_build.sh` too.

I chose `ARROW_USE_CCACHE:=OFF` here because `cpp_build.sh` uses it. I set `ARROW_USE_CCACHE=ON` in caller like other jobs.